File: src/main/java/com/appointmed/model/Appointment.java
package com.appointmed.model;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Table(name = "appointments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

    public enum Status { PENDING, CONFIRMED, COMPLETED, CANCELLED, NO_SHOW }
    public enum PaymentStatus { UNPAID, PAID, REFUNDED }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "patient_id", nullable = false)
    private User patient;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "doctor_id", nullable = false)
    private Doctor doctor;

    @Column(nullable = false)
    private LocalDate appointmentDate;

    @Column(nullable = false)
    private LocalTime appointmentTime;

    private LocalTime endTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private Status status = Status.PENDING;

    @Column(length = 500)
    private String reason;

    @Lob
    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(precision = 10, scale = 2)
    @Builder.Default
    private BigDecimal fee = BigDecimal.ZERO;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private PaymentStatus paymentStatus = PaymentStatus.UNPAID;

    @Column(updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: mobile/app/src/main/java/com/appointmed/mobile/data/model/Appointment.kt
package com.appointmed.mobile.data.model

data class Appointment(
    val id: Long = 0,
    val patient: AppointmentUser? = null,
    val doctor: AppointmentDoctor? = null,
    val appointmentDate: String = "",
    val appointmentTime: String = "",
    val endTime: String? = null,
    val reason: String? = null,
    val fee: Double = 0.0,
    val status: String = "PENDING"  // PENDING, CONFIRMED, COMPLETED, CANCELLED
)

data class AppointmentUser(
    val id: Long = 0,
    val name: String = "",
    val email: String = "",
    val avatarData: String? = null,
    val avatarUrl: String? = null
)

data class AppointmentDoctor(
    val id: Long = 0,
    val specialization: String = "",
    val phone: String? = null,
    val clinicAddress: String? = null,
    val avatarUrl: String? = null,
    val user: AppointmentUser? = null
)
